presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/SimplifyCardinalityMapRewriter.java

```diff
@@ -76,10 +91,14 @@ public RowExpression rewriteCall(CallExpression node, Void context, RowExpressio
         private RowExpression newFunctionIfRewritten(CallExpression node, List<RowExpression> rewrittenArguments)
         {
             if (!node.getArguments().equals(rewrittenArguments)) {
+                List<Type> types = rewrittenArguments.stream()
+                        .map(RowExpression::getType)
+                        .collect(toImmutableList());
+                FunctionHandle mapCardinalityFunctionHandle = functionAndTypeManager.lookupFunction("cardinality", fromTypes(types));
```

Comment:
This function gets called even function is not cardinality(). So lets fix it above near line 77.

Reply:
got it... maybe i should resolve all function handles then and not just cardinality, to be extra safe.

Reply:
we should test a case like `cardinality(map_keys(x))+cardinality(map_keys(x))`

Reply:
Some tests in presto-main also failed here